tests/session/cmd/test_parallel.py

```diff
@@ -141,7 +141,7 @@ def test_keyboard_interrupt(tox_project: ToxProjectCreator, demo_pkg_inline: Pat
     )
     cmd = ["-c", str(proj.path / "tox.ini"), "p", "-p", "1", "-e", f"py,py{sys.version_info[0]},dep"]
     process = Popen([sys.executable, "-m", "tox", *cmd], stdout=PIPE, stderr=PIPE, universal_newlines=True)
-    while not marker.exists():
+    while not marker.exists() and (process.poll() is None):
```

Comment:
While in a bit of broken state, this test was hanging for me. Suggesting that we use [`Popen.poll`](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.poll) to check if the child process has already terminated (likely with an error if marker does not exist).